Scripts/Engine/CustomBar.lua

```diff
@@ -0,0 +1,848 @@
+--	local blendMode = TEN.Effects.BlendID.ALPHATEST
+--	local textPos = TEN.Vec2(posBar.x, posBar.y+10)
+--
+--	-- This function creates the bar and displays it
+--	CustomBar.Create("Test", 50, 1000,
+--		ObjID, 2, colorBar, posBar, 0, scaleBar, alignMode, scaleMode, blendMode,
+--		ObjID, 4, colorBar, posBG, 0, scaleBG, alignMode, scaleMode, blendMode,
+--		"Test 1", textPos, {}, 1, colorBar, false, 50, true, 0.25)
+--
+--		-- This method displays the bar
+--		bar:SetVisibility(true)
+--	end
+--
+-- @luautil CustomBar
+
+
+local CustomBar = {}
+
+CustomBar.__index = CustomBar
+
+LevelFuncs.Engine.CustomBar = {}
+LevelVars.Engine.CustomBars = {bars = {}, EnemiesHpBar = {status = nil}}
+
+---
+-- Creates a custom progress bar with extensive configuration options.
+-- Parameters:
+-- @tparam string barName Unique name for the bar.
+-- @tparam float startvalue Initial value of the bar.
+-- @tparam float maxvalue Maximum value of the bar.
+-- @tparam Objects.ObjID objectIDbg Object ID for the bar's background sprite.
+-- @tparam number spriteIDbg SpriteID from the specified object for the bar's background.
+-- @tparam Color colorbg Color of bar's background.
+-- @tparam Vec2 posBG X,Y position of the bar's background in screen percent (0-100).
+-- @tparam float rotBG rotation of the bar's background. sprite (0-360).
+-- @tparam Vec2 scaleBG X,Y Scaling factor for the bar's background sprite.
+-- @tparam View.AlignMode alignModebg Alignment for the bar's background.
+-- @tparam View.ScaleMode scaleModebg Scaling for the bar's background.
+-- @tparam Effects.BlendID blendModebg Blending modes for the bar's background.
+-- @tparam Objects.ObjID objectIDbar Object ID for the bar sprite.
+-- @tparam number spriteIDbar SpriteID from the specified object for the bar.
+-- @tparam Color colorbar Color of the bar.
+-- @tparam Vec2 posBar X,Y position of the bar in screen percent (0-100).
+-- @tparam float rot rotation of the bar's sprite (0-360).
+-- @tparam Vec2 scaleBar X,Y Scaling factor for the bar's sprite.
+-- @tparam View.AlignMode alignMode Alignment for the bar.
+-- @tparam View.ScaleMode scaleMode Scaling for the bar.
+-- @tparam Effects.BlendID blendMode Blending modes for the bar.
+-- @tparam string text Text to display on the bar.
+-- @tparam Vec2 textPos X,Y position of the text.
+-- @tparam Strings.DisplayStringOption textOptions alignment and effects for the text. Default: None. Please note text is automatically aligned to the LEFT
+-- @tparam number textScale Scale factor for the text.
+-- @tparam Color textColor Color of the text.
+-- @tparam bool hideText Whether to hide the text.
+-- @tparam number alphaBlendSpeed Speed of alpha blending for bar visibility (0-255).
+-- @tparam bool blink Whether the bar blinks.
+-- @tparam number blinkLimit % Limit below which bar starts blinking (0-1).
+--
+-- @treturn CustomBar The custombar in its hidden state
+--
+CustomBar.Create = function (barName, startvalue, maxvalue,

```

Comment:
As I said earlier in Discord, having 27 arguments in a function is not very good. I recommend to pack all arguments to a table.

Reply:
Done